pkg/buildkit/buildkit.go

```diff
@@ -354,19 +377,49 @@ func (c *Client) runSolve(ctx context.Context, so bkclient.SolveOpt) error {
 		return err
 	})
 
+	var size int64
+
 	eg.Go(func() error {
-		if _, err := c.bk.Solve(ctx, nil, so, ch); err != nil {
+		res, err := c.bk.Solve(ctx, nil, so, ch)
+		if err != nil {
 			return err
 		}
 
 		c.log.Info("Solve complete")
+		imageName := res.ExporterResponse["image.name"]
+		ref, err := name.ParseReference(imageName)
+		if err != nil {
+			return err
+		}
+		registryName := ref.Context().RegistryStr()
+		auth, err := c.ResolveAuth(registryName)
```

Comment:
Yeah, the thinking here was that - here, `buildkit.Solve` returns a response, but that response unfortunately doesn't include the final image size - so we use the image url and calculate the image size to return (sum the layers).

For reference, the solve response looks approximately like
```
{
    "containerimage.config.digest": ...,
    "containerimage.descriptor": {
        "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
        "digest": "...",
        "size": 2267,
        "platform": {
            "architecture": "amd64",
            "os": "linux"
        }
    },
    "containerimage.digest": "...",
    "image.name": "..."
}
```

Reply:
Good point, I was also worried about adding the new dependencies into Heph - I think I'll schedule a quick call after hackathon for the team to think through these different options.

Re: failure modes - that makes sense, just pushed a change that improves this a little bit https://github.com/dominodatalab/hephaestus/pull/131/files#diff-9bf6bd7b5671bcb828a58b3c5ef62d040d8ded0f7dc901e2b4c01191cb025dd4R418-R421 - the errors when calculating image size should _**not**_ count as a build error (it's metadata) - instead, this will log any errors and still return "build success".

TIL about the multi-arch nuance, I'll have to think about that.